src/plugins/opensearch_ui_shared/__packages_do_not_import__/json_editor_with_diagnostics_options/json_editor_with_diagnostics_options.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React, { useRef, useEffect } from 'react';
+import { monaco } from '@osd/monaco';
+
+export function useDiagnosticsOptions(props: {
+  monaco?: typeof monaco;
+  diagnosticsOptions?: monaco.languages.json.DiagnosticsOptions;
+}) {
+  useEffect(() => {
+    const oldOptionsSettings = props.monaco?.languages.json.jsonDefaults.diagnosticsOptions;
+    if (props.monaco) {
+      props.monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
+        ...oldOptionsSettings,
+        ...props.diagnosticsOptions,
+        schemas: [
+          ...(oldOptionsSettings?.schemas || []),
+          ...(props.diagnosticsOptions?.schemas || []),
+        ],
+      });
+    }
+    return () => {
+      props.monaco?.languages.json.jsonDefaults.setDiagnosticsOptions(oldOptionsSettings || {});
+    };
+  }, [props.monaco, props.diagnosticsOptions]);
+}
+
+const jsonCode = ['{', '    "p1": "v3",', '    "p2": false', '}'].join('\n');
+const modelUri = monaco.Uri.parse('a://b/foo.json');
+const model = monaco.editor.createModel(jsonCode, 'json', modelUri);
+
+export function JsonEditorWithDiagnosticsOptions() {
+  const containerRef = useRef<HTMLDivElement>(null);
+  useDiagnosticsOptions({
+    monaco,
+    diagnosticsOptions: {
+      validate: true,
+      schemas: [
+        {
+          uri: 'http://myserver/foo-schema.json',
```

Comment:
Dummy links in tests that don't resolve need to be added to https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/.lycheeexclude